### DISCUSSION TITLE
IBM I code coverage command not found

### DISCUSSION THREAD

**Discussion by m-tyler** [codefori/vscode-ibmi#782] [Q&A]
I receive this error with the latest code coverage version.
**Code Coverage for IBM i v0.1.3**

```
Error running command code-coverage-ibmi.runCoverageTest: command 'code-coverage-ibmi.runCoverageTest' not found. This is likely caused by the extension that contributes code-coverage-ibmi.runCoverageTest.
```

I am placing here because there isn't an issue report option on the code coverage project or discussion board. 

Looking in the installed extensions for the code coverage, the command is referenced in the package.json file.    However, I am not find a register for the above command in the other sources. 

I uninstalled, restarted and reinstalled from extension package **IBM i Development Pack v0.0.8** without any change.

**Comment by m-tyler** [comment]
I also noticed that when I attempt to create a test using the right click on a member, it is not filling out the inputs with the member information but parsing out the IASP name of our system.

![image](https://user-images.githubusercontent.com/1353665/181388024-a642fb32-8a38-4b4a-9219-965a44cda918.png)

**Reply by worksofliam** [reply]
I can look into those issues. I would recommend you find the codecov command on your system and make sure the library it's in is part of your library list. Possibly QDEVTOOLS?

**Reply by m-tyler** [reply]
Yes, we have QDEVTOOLS installed and a proxy command pointing to QDEVTOOLS/CODECOV.  I can take the command that is supposed to be run by the tool and run it on our IBM system and I get a results file. 

I just don't seem to be able to trigger off the `run` feature of the Code coverage for IBM i command feature. 

The Code Coverage extension does find the results file (if I trick it a bit) but I am getting mixed results from within the results file that i need to look into and are not part of this extension.  View from within RDi the results are odd to me.

**Reply by worksofliam** [reply]
@m-tyler did you try adding whichever library it exists in to your library list inside of vscode before running code coverage?

**Reply by m-tyler** [reply]
First attempt I did not have QDEVTOOLS added to my active settings.  It reported back an error about not being able to find message file in QDEVTOOLS. I added it, saved the profile to the one I was working on, retried running test.  This time I received another errors, sorry I dont have that one handy.

I restarted VSCODE, reconnected to my system and tried to run the test again, through the Code Coverage view, through the command pallet and using F1.     

The command used is logged to the output so I copied that to a 5250 command line and it started running (after adding QDEVTOOLS to the *LIBL).

**Reply by worksofliam** [reply]
@m-tyler 

> ... and it started running (after adding QDEVTOOLS to the *LIBL).

you mean from the Code Coverage view?

**Reply by m-tyler** [reply]
No, sorry, I meant to show that we have CODECOV installed and working on our IBM i by running the same command on a 5250 command line.    I had to add QDEVTOOLS to the 5250 session before running the CODECOV command that should have run through VS CODE.

**Comment by worksofliam** [comment]
I was able to create a test and run it quite fine. Ensure QDEVTOOLS is on your library list (as it is in my user library list)

<img width="1238" alt="image" src="https://user-images.githubusercontent.com/3708366/181389602-4c78705f-50c3-4ddd-abdd-1d158383eaac.png">

**Reply by worksofliam** [reply]
I am looking into the ASP issue now.

**Reply by worksofliam** [reply]
Fixed it. I will do a release tonight.

**Reply by m-tyler** [reply]
Thanks for the quick turn around. I'll try again tomorrow.

**Comment by worksofliam** [comment]
@m-tyler Code Coverage is coming out with 0.1.5 today. Give that a try when it's available.

**Reply by m-tyler** [reply]
I see that adding a test from the member right-click is working but the Run Test command is still not found.  I did notice that for newly started tests or newly activated sessions to an IBM I, the test will run but not again until I restart VS Code.   Subsequent attempts to run the test I receive the error posted in the OP. 

So I can see that CODECOV does work but in a limited form.  I have an issue with the way my program runs that I need to solve before I ask/submit anything else.  It seems as if my parameters passed are not used when run through VS Code but are used when run from a 5250 command line.

**Reply by worksofliam** [reply]
@m-tyler Are you able to record your screen so I can see the issues you are facing? Or perhaps we can setup a call to take a look?

**Reply by m-tyler** [reply]
The issue I was trying to overcome is the program was ending too quickly.  It looked like my parameters I passed on the call were not being used but that is not the case.  The SSH jobs are using the job date of the original SSH service.  I believe this because I added code to dump the program at a certain point and *DATE is set to 07132022.

I have a *JOB date variable I create and it has the same value.

![image](https://user-images.githubusercontent.com/1353665/181568228-393ce0ec-fd3d-4192-aebd-c33c868194e3.png)

I looked at the SBS SSHSBS (where the Code for IBM I ssh jobs end up) and all the jobs have their job dates of 07/13/2022.  This seems like a flaw on IBM's side but it is what it is and I don't have a workaround at the moment. 


  ---------------------------------------------------------------------------------------------------------------------
On the part of the test not being runnable subsequent times during the same session, I found out that if I remove the test results file it will run again.   So are tests supposed to be capable of running multiple times per session?  
  steps to recreate:
  1. Create a test (any method is fine)
  2. Open Code Coverage View
  3. Run the test
  4. Right-click on the test to have it run again.  
  5. You will receive the error posted in the OP.

**Reply by m-tyler** [reply]
Okay, I have to expose some more of my ignorance.  I just noticed that the refresh icon is actually a run action on all the defined tests.  I am not certain that is a beneficial feature.   When using RDi I would set up many tests and leave them there in case I want to return to them.   I wouldn't want to have a test re-run unless I want it to. 

The right-click run command still does not function.

**Reply by m-tyler** [reply]
Playing around with the tests I created more than one test.  After a bit of time, I decided to remove one of the tests.  Upon completing that task, the tests were re-run without me asking for it.  I think the re-run feature needs to be a specific ask by the user.
